### PR TITLE
uml: avoid unnecessary repetition

### PR DESCRIPTION
**Pull request by fogti**

### CODE REVIEW

**Comment by drupol** [comment]
Hey hi !

Thanks for your proposal.

Your PR proposes to remove the duplicated part as such:



The current diagram is:

 has indeed two similar parts that are almost the same.

In the beginning, I also tried to remove the duplicated parts, but while doing it, I noticed that they are actually different.
The first part "Trigger CI" and then "Add commit(s)", while the second one "Add commit(s)" and then "Trigger CI". This is two different things.

I'm not the expert here and I do not close the door, leaving this PR open to gather some feedback from other people and we'll merge it eventually.

**Comment by drupol** [comment]
I made some local tests and I'm ok to merge this PR, but you would need to add an extra step.

Basically, you just need to add `backward :Add commit(s);` before the last `repeat while...` and it will be ok.

Result:
image

Do you want to make the changes?

File: src/life-of-a-pr.plantuml
Comment (drupol) [inline src/life-of-a-pr.plantuml:30]:
```suggestion
  |Contributor|
  backward :Add commit(s);
  repeat while (Changes requested?) is (yes) not (no)
```

**Review by drupol — CHANGES_REQUESTED** [review]
I suggest this minor change to make your PR valid.

Also,  a reorganisation of the swimlanes as seen in the screenshot I posted and we're good to go, but I can do it myself after this PR is merged if you prefer.

**Comment by drupol** [comment]
Thanks for your contribution.

**Comment by fogti** [comment]
I think another step is missing, because usually long lists of changes are squashed before stuff gets merged.

**Comment by drupol** [comment]
> I think another step is missing, because usually long lists of changes are squashed before stuff gets merged.

A new PR ? :)

**Comment by drupol** [comment]
> I think another step is missing, because usually long lists of changes are squashed before stuff gets merged.

After giving a second though, I think this is part of the "Waiting for approvals" part of the workflow.

**Pull request merged**
